aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/auth/credentials/StaticCredentialsProviderNative.kt

```diff
@@ -8,19 +8,10 @@ package aws.sdk.kotlin.crt.auth.credentials
 /**
  * A credentials provider for a fixed set of credentials
  */
-public actual class StaticCredentialsProvider internal actual constructor(builder: StaticCredentialsProviderBuilder) :
-    CredentialsProvider {
+public actual class StaticCredentialsProvider internal actual constructor(private val builder: StaticCredentialsProviderBuilder) : CredentialsProvider {
     public actual companion object {}
-
-    override suspend fun getCredentials(): Credentials {
-        TODO("Not yet implemented")
-    }
-
-    override fun close() {
-        TODO("Not yet implemented")
-    }
-
-    override suspend fun waitForShutdown() {
-        TODO("Not yet implemented")
-    }
+    override suspend fun getCredentials(): Credentials =
+        Credentials(builder.accessKeyId!!, builder.secretAccessKey!!, builder.sessionToken)
+    override fun close() { }
+    override suspend fun waitForShutdown() { }
 }
```

Comment:
**Correctness**: Typically, retaining a reference to a mutable builder is a bad idea since it means the values used by your supposedly-static provider can be changed after the fact:

```kotlin
val builder = StaticCredentialsProviderBuilder().apply {
    accessKeyId = "foo"
    secretAccessKey = "bar"
}
val built = builder.build()
println(built.getCredentials()) // foo/bar

builder.secretAccessKey = "baz"
println(built.getCredentials()) // foo/baz
```

We should either create the credentials once at initialization time or copy the builder values to local private members.